### ISSUE TITLE
Unable to create model in fission version 1.20.4 when upgrading from 1.20.1

### ISSUE THREAD

**Issue by rajat42059** [fission/fission#3023]
**Fission Version:**
client:
  fission/core:
    BuildDate: "2024-09-02T07:18:37Z"
    GitCommit: cf55fbec
    Version: v1.20.4
server:
  fission/core:
    BuildDate: "2024-09-02T07:18:37Z"
    GitCommit: cf55fbec
    Version: v1.20.4

**Kubectl version**

Client Version: v1.26.3
Kustomize Version: v4.5.7
Server Version: v1.28.9

**To Reproduce**

Current version of fission installed: 1.20.1
New Version Installed : 1.20.4

When upgrading the fission version from 1:20.1->1.20.4

Model creation is working fine on 1.20.1

when updated fto 1.20.4 got the following error in router:-
2024-09-20T08:55:54.824961861Z {"level":"info","ts":"2024-09-20T08:55:54.824Z","caller":"crd/client.go:143","msg":"Checking function CRD access","namespace":"default","timeout":"30s"}
2024-09-20T08:55:55.909738766Z {"level":"info","ts":"2024-09-20T08:55:55.909Z","caller":"router/router.go:217","msg":"starting router","port":8888}
2024-09-20T08:55:55.909893968Z {"level":"info","ts":"2024-09-20T08:55:55.909Z","caller":"httpserver/server.go:22","msg":"starting server","service":"router/metrics","addr":":8080"}
2024-09-20T08:55:55.918949597Z {"level":"info","ts":"2024-09-20T08:55:55.909Z","caller":"httpserver/server.go:22","msg":"starting server","service":"router","addr":":8888"}
2024-09-20T09:36:39.124555051Z 2024/09/20 09:36:39 [DEBUG] POST http://executor.fission/v2/getServiceForFunction
2024-09-20T09:36:51.628764549Z 2024/09/20 09:36:51 [DEBUG] POST http://executor.fission/v2/getServiceForFunction
2024-09-20T09:38:39.112724503Z 2024/09/20 09:38:39 [ERR] POST http://executor.fission/v2/getServiceForFunction request failed: Post "http://executor.fission/v2/getServiceForFunction": context deadline exceeded
2024-09-20T09:38:39.115076240Z {"level":"error","ts":"2024-09-20T09:38:39.112Z","logger":"triggerset.http_trigger_set.e52bbe-http","caller":"router/functionHandler.go:653","msg":"error from GetServiceForFunction","trace_id":"8f8a7ae09bd8195da651597327d857b6","error":"error posting to getting service for function: POST http://executor.fission/v2/getServiceForFunction giving up after 1 attempt(s): context deadline exceeded","errorVerbose":"POST http://executor.fission/v2/getServiceForFunction giving up after 1 attempt(s): context deadline exceeded\nerror posting to getting service for function\ngithub.com/fission/fission/pkg/executor/client.(*client).GetServiceForFunction\n\tpkg/executor/client/client.go:94\ngithub.com/fission/fission/pkg/router.functionHandler.getServiceEntryFromExecutor\n\tpkg/router/functionHandler.go:650\ngithub.com/fission/fission/pkg/router.functionHandler.getServiceEntry\n\tpkg/router/functionHandler.go:674

**Executor logs:-**

{"level":"info","ts":"2024-09-23T10:38:04.818Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:113","msg":"creating pool","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-198406-1f04db"}}
{"level":"info","ts":"2024-09-23T10:38:04.818Z","logger":"generic_pool_manager.pool_pod_controller","caller":"poolmgr/poolpodcontroller.go:321","msg":"created pool for the environment","env":"python-env-build-144900-95ee40","namespace":"default"}
{"level":"info","ts":"2024-09-23T10:38:05.122Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp_deployment.go:233","msg":"deployment created","deployment":"poolmgr-python-env-build-1-default-538367009","ns":"fission-function","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-198406-1f04db"}}
{"level":"info","ts":"2024-09-23T10:38:05.122Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/readyPodController.go:49","msg":"readyPod controller started","env":"python-env-build-198406-1f04db","envID":"56436248-9065-443c-9162-a5f8e19e6e42"}
{"level":"info","ts":"2024-09-23T10:38:05.122Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:113","msg":"creating pool","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-215465-0b9da2"}}
{"level":"info","ts":"2024-09-23T10:38:05.122Z","logger":"generic_pool_manager.pool_pod_controller","caller":"poolmgr/poolpodcontroller.go:321","msg":"created pool for the environment","env":"python-env-build-198406-1f04db","namespace":"default"}
{"level":"info","ts":"2024-09-23T10:38:05.420Z","caller":"httpserver/server.go:22","msg":"starting server","service":"executor/metrics","addr":":8080"}
{"level":"info","ts":"2024-09-23T10:38:05.421Z","logger":"executor","caller":"httpserver/server.go:22","msg":"starting server","service":"executor","addr":":8888"}
{"level":"info","ts":"2024-09-23T10:38:05.422Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp_deployment.go:233","msg":"deployment created","deployment":"poolmgr-python-env-build-2-default-535606343","ns":"fission-function","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-215465-0b9da2"}}
{"level":"info","ts":"2024-09-23T10:38:05.422Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/readyPodController.go:49","msg":"readyPod controller started","env":"python-env-build-215465-0b9da2","envID":"af92bce6-a5e2-46f7-854f-db823daf830f"}
{"level":"info","ts":"2024-09-23T10:38:05.422Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp.go:113","msg":"creating pool","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-143593-97cd15"}}
{"level":"info","ts":"2024-09-23T10:38:05.422Z","logger":"generic_pool_manager.pool_pod_controller","caller":"poolmgr/poolpodcontroller.go:321","msg":"created pool for the environment","env":"python-env-build-215465-0b9da2","namespace":"default"}
{"level":"info","ts":"2024-09-23T10:38:05.728Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/gp_deployment.go:233","msg":"deployment created","deployment":"poolmgr-python-env-build-1-default-563658646","ns":"fission-function","environment":{"apiVersion":"fission.io/v1","kind":"Environment","namespace":"default","name":"python-env-build-143593-97cd15"}}
{"level":"info","ts":"2024-09-23T10:38:05.728Z","logger":"generic_pool_manager.generic_pool","caller":"poolmgr/readyPodController.go:49","msg":"readyPod controller started","env":"python-env-build-143593-97cd15","envID":"4195575e-55e1-43ca-ba96-6b0150903ff9"}
{"level":"info","ts":"2024-09-23T10:38:05.728Z","logger":"generic_pool_manager.pool_pod_controller","caller":"poolmgr/poolpodcontroller.go:321","msg":"created pool for the environment","env":"python-env-build-143593-97cd15","namespace":"default"}

**Comment by rajat42059** [comment]
@sanketsudake..we noticed when we updated to 1.20.4..there is no pods in fission-function..

![Uploading image.png…]()

**Comment by sanketsudake** [comment]
Please share your helm configuration 

```
helm get values <fission_installation> -n <fission installed _namespace>
```

Also share share specs for function and enviroment

```
fission fn list 
fission env list 
```

I think there may be something wrong at configuration level, which was not supported but being used.

**Comment by rajat42059** [comment]
@sanketsudake  
**helm get values fission -n fission**

USER-SUPPLIED VALUES:
analytics: false
builderNamespace: fission-builder
defaultNamespace: default
functionNamespace: fission-function
prometheus:
  enabled: false
  serviceEndpoint: https://prometheus-kube-prometheus-prometheus.operations.svc.cluster.local
routerServiceType: ClusterIP

**fission env list** 

<img width="955" alt="image" src="https://github.com/user-attachments/assets/8a5063ef-4671-43f1-9330-5b50998de5c3">


**fission fn list** 



<img width="713" alt="image" src="https://github.com/user-attachments/assets/d60473cd-50ce-4c97-8104-50484c354fca">

**Comment by rajat42059** [comment]
@sanketsudake...any update on this?

**Comment by soharab-ic** [comment]
**Analysis:**
During upgrade from fission `v1.20.1` to `v1.20.4`, if cross namespaces are used like `fission-builder` and `fission-function` for builder and function namespace respectively then the poolmgr deployment is not created.
If we look at the logs of executor, we find that executor cleans up the poolmgr deployment and then recreates it. Althogh, there are no error while recreating the poolmgr deployment.
We still observe that there are no poolmgr deployment in `fission-function` namespace.
The root cause of this issue is the OwnerReferences set to K8s resources created by Fission in this PR.
https://github.com/fission/fission/pull/2964

**Workaround:**
Avoid using cross namespace , ie,  builder and environment pods should be preferred to be created in the same namespace.
This can be done by not setting builder namespace and function namespace.
```
builderNamespace: ""
functionNamespace: ""
```
Fission would be deprecating and removing cross namespace access eventually.

**Fix:**
We would be adding `DISABLE_OWNER_REFERENCES` env variable to executor which would be false by default. If user is using cross namespace access then they would need it set it true.
We would add support to set disable_owner_reference to true via helm chart.
So,  we would support existing behaviour in short term but eventually we would get rid of following.
```
builderNamespace: ""
functionNamespace: ""
disable_owner_reference: 
```

**Comment by rajat42059** [comment]
@soharab-ic ..thanks for the input..we will try settings 

builderNamespace: ""
functionNamespace: ""

will update the status here 

also, when the new version with  above changes is planned?

**Comment by sanketsudake** [comment]
Short term fix available in latest release 1.20.5
